<?xml version="1.0" encoding="ISO-8859-15"?>
<launch>
  <arg name="show_frontend" default="false"/>
  <arg name="dsg_topic" default="incremental_dsg_builder_node/dsg" unless="$(arg show_frontend)"/>
  <arg name="dsg_mesh_topic" default="incremental_dsg_builder_node/dsg_mesh" unless="$(arg show_frontend)"/>
  <arg name="dsg_topic" default="incremental_dsg_builder_node/frontend/dsg" if="$(arg show_frontend)"/>
  <arg name="dsg_mesh_topic" default="incremental_dsg_builder_node/frontend/dsg_mesh" if="$(arg show_frontend)"/>

  <arg name="world_frame" value="world"/>

  <arg name="viz_config_dir" default="$(find hydra_utils)/config/hydra_visualizer"/>
  <arg name="visualizer_ns" default="/hydra_dsg_visualizer"/>

  <arg name="start_rviz" default="true"/>
  <arg name="rviz_dir" default="$(find hydra_utils)/rviz"/>
  <arg name="rviz_file" default="hydra_streaming_visualizer.rviz"/>
  <arg name="rviz_path" default="$(arg rviz_dir)/$(arg rviz_file)"/>

  <arg name="verbosity" default="0"/>

  <include file="$(find hydra_utils)/launch/includes/hydra_visualizer_params.xml">
      <arg name="viz_config_dir" value="$(arg viz_config_dir)"/>
      <arg name="visualizer_ns" value="$(arg visualizer_ns)"/>
      <arg name="world_frame" default="$(arg world_frame)"/>
  </include>

  <node pkg="hydra_utils" type="hydra_visualizer_node" name="hydra_visualizer_node"
        output="log"
        required="true"
        args="-alsologtostderr -colorlogtostderr -v=$(arg verbosity)">
    <param name="visualizer_ns" value="$(arg visualizer_ns)"/>
    <param name="load_graph" value="false"/>

    <param name="mesh_plugin_type" value="PGMO"/>
    <param name="scene_graph_filepath" value=""/>
    <param name="visualizer_ns" value="$(arg visualizer_ns)"/>
    <param name="mesh_plugin_ns" value="dsg_mesh"/>
    <param name="output_path" value=""/>

    <remap from="~dsg" to="$(arg dsg_topic)"/>
    <remap from="~dsg_mesh_updates" to="$(arg dsg_mesh_topic)"/>
  </node>

  <node name="rviz" pkg="rviz" type="rviz" output="log" if="$(arg start_rviz)" args="-d $(arg rviz_path)"/>

</launch>
